server/services/prisonerSearchService.ts

```diff
@@ -14,6 +15,50 @@ export default class PrisonerSearchService {
     private readonly prisonerSearchStore: PrisonerSearchStore,
   ) {}
 
+  /**
+   * Returns the [PrisonerSummary]s for the specified prisonId
+   *
+   * The HMPPS `prisoner-search` API is a paged API. This function calls the API starting from page 0 until there are no
+   * more pages remaining. The cumulative array of Prisoner records from all API calls are mapped and
+   * grouped into an array of `PrisonerSummary` within the returned `PrisonerSummaries` object.
+   */
+  async getPrisonersByPrisonId(prisonId: string, username: string): Promise<PrisonerSummaries> {
```

Comment:
This is the main objective of this PR - to create a proper service method in the service class (should have been there before 🙄 ) to call prisoner-search to get all prisoners in a prison, and to also account for the fact that it is a paged API (previously we set a page size of 9999 and assumed it would always come back in one page!)